internal/hcp/registry/types.bucket.go

```diff
@@ -469,3 +495,115 @@ func (b *Bucket) HeartbeatBuild(ctx context.Context, build string) (func(), erro
 		close(heartbeatChan)
 	}, nil
 }
+
+func (b *Bucket) StartBuild(ctx context.Context, buildName string) error {
```

Comment:
With these method names does the Bucket type also implement the Refistry interface?

if it does and that is not intentional we should probably give these different names. Maybe something like MarkBuildAsStarted etc …

Reply:
Well, that's true, as it stands Bucket is an implementation of registry.
This has the side-effect of having to rename the current `CompleteBuild` to `completeBuild`, as it does the update of the status to `DONE` and uploading the labels and other metadata. This can indeed be renamed in `Bucket` so it's not an implementation of the `Registry` interface.
I'll do another reroll later for that

Reply:
Note: since the two are in the same package, we can make the `Bucket` functions private, so it's not an implementation anymore. This removes the potential confusion when looking for the Registry implementations.

Reply:
Sounds good to me. 🚢